test/index.spec.ts

```diff
@@ -1058,3 +1065,10 @@ describe('helpers', () => {
     })
   })
 })
+
+describe('unknown protocols', () => {
+  it('throws an error', () => {
+    const str = '/ip4/127.0.0.1/unknown'
+    expect(() => multiaddr(str)).to.throw()
```

Comment:
It's a good idea to assert something about the thrown error in order to ensure the correct error was thrown, something like:

```suggestion
    expect(() => multiaddr(str)).to.throw()
      .with.property('message').that.includes('something specific')
```